### DISCUSSION TITLE
How to use a component within a code block?

### DISCUSSION THREAD

**Discussion by didvc** [mdx-js/mdx#2528] [Q&A]
How to use a component within a code block in mdx?

e.g.

    ```
    console.log("Hello, <Username />")
    ```

**Comment by wooorm** [comment]
You cannot, you should not, this will never be done.

Use JSX:

````mdx
<pre><code>console.log("Hello, &lt;Username /&gt;")
</code></pre>
````

**Reply by remcohaszing** [reply]
I interpreted the question as wanting to do this:

```mdx
<pre><code>console.log("Hello, <Username />")
</code></pre>
```

I hope having multiple interpretations of the OP helps clarify why supporting components inside MDX code blocks is too ambiguous.

**Reply by didvc** [reply]
The example is just a simplification for easier understanding. 
In my case, what I wanted to is to use Tailwind `blur` div within the code block. I'm working on a site which has paid content area, including code snippets, so I needed it. 
I hope this helps.

**Reply by remcohaszing** [reply]
You may be interested in https://codehike.org/
